server/auth/store.go

```diff
@@ -935,6 +939,8 @@ func NewAuthStore(lg *zap.Logger, be AuthBackend, tp TokenProvider, bcryptCost i
 
 	be.CreateAuthBuckets()
 	tx := be.BatchTx()
+	// We should call LockOutsideApply here, but the txPostLockHoos isn't set
```

Comment:
Let's just call `LockOutsideApply`

Reply:
It's authBatchTx instead of the batchTx, see [auth.go#L108](https://github.com/ahrtr/etcd/blob/4d1070af9e835b5dc7bb6f5924f663a5b2c10973/server/storage/schema/auth.go#L108).

Reply:
Majority calls of` (atx *authBatchTx) Lock() ` are coming from the apply workflow, but this one is a special case. 

Please note that I enhanced the authBackend to support ReadTx (previously it only supports BatchTx), and all readonly operations go to the ReadTx, all writing operations go to the BatchTx.